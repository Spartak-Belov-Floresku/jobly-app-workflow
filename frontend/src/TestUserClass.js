import GlobalContext from "./helper/GlobalContext";

const mockUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
};

const TestUserClass = ({ children, currUser = mockUser, jobHasBeenApplied = () => false }) => (
  <GlobalContext.Provider value={{ currUser, jobHasBeenApplied }}>
    {children}
  </GlobalContext.Provider>
);

export { TestUserClass };