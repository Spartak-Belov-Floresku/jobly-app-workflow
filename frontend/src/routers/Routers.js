import { Switch, Route, Redirect } from "react-router-dom";
import Home from "../home/Home";
import SignUpForm from "../forms/SignUpForm";
import LoginForm from "../forms/LoginForm";
import LoginRoutes from "./LoginRoutes";
import Companies from "../companies/Companies";
import CompanyDetails from "../companies/CompanyDetails";
import GetAllJobs from "../jobs/GetAllJobs";
import UserForm from "../forms/UserForm";

/** Site-wide routes.
 *
 * Parts of site should only be visitable when logged in. Those routes are
 * wrapped by <LoginRoutes>, which is an authorization component.
 *
 * Visiting a non-existant route redirects to the homepage.
 */

function Routes({ Signup, Login }) {
  
  return (
      <div className="pt-5">
        <Switch>

          <Route exact path="/">
            <Home />
          </Route>

          <Route exact path="/login">
            <LoginForm login={Login} />
          </Route>

          <Route exact path="/signup">
            <SignUpForm signup={Signup} />
          </Route>

          <LoginRoutes exact path="/companies">
            <Companies />
          </LoginRoutes>

          <LoginRoutes exact path="/companies/:name">
            <CompanyDetails />
          </LoginRoutes>

          <LoginRoutes exact path="/jobs">
            <GetAllJobs />
          </LoginRoutes>

          <LoginRoutes exact path="/profile">
            <UserForm />
          </LoginRoutes>

          <Redirect to="/" />
        </Switch>
      </div>
  );
}

export default Routes;
