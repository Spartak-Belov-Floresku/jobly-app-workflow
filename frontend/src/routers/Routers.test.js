import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { TestUserClass } from "../TestUserClass";
import Routers from "./Routers";

it("renders without crashing logged in mock user", () => {
  render(
      <MemoryRouter>
        <TestUserClass>
          <Routers />
        </TestUserClass>
      </MemoryRouter>,
  );
});

it("matches snapshot logged in user", () => {
  const { asFragment } = render(
      <MemoryRouter>
        <TestUserClass>
          <Routers />
        </TestUserClass>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", () => {
  const { asFragment } = render(
      <MemoryRouter>
        <TestUserClass currentUser={null}>
          <Routers />
        </TestUserClass>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});