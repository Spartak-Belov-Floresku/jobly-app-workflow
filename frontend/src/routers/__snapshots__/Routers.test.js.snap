// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot logged in user 1`] = `
<DocumentFragment>
  <div
    class="pt-5"
  >
    <div
      class="Homepage"
    >
      <div
        class="container text-center"
      >
        <h1
          class="mb-4 font-weight-bold"
        >
          Jobly
        </h1>
        <p
          class="lead"
        >
          All the jobs in one, convenient place.
        </p>
        <h2>
          Welcome Back, testuser!
        </h2>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="pt-5"
  >
    <div
      class="Homepage"
    >
      <div
        class="container text-center"
      >
        <h1
          class="mb-4 font-weight-bold"
        >
          Jobly
        </h1>
        <p
          class="lead"
        >
          All the jobs in one, convenient place.
        </p>
        <h2>
          Welcome Back, testuser!
        </h2>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
